### PR TITLE
WT-13262 Limit the number of btrees processed for obsolete time window info in eviction (4.4)

### PR DESCRIPTION
**Pull request by etienneptl**

### CODE REVIEW

File: src/evict/evict_page.c [hunk 1]
```diff
@@ -567,34 +567,40 @@ static int
...
+
     /*
      * Pages that the application threads are evicting should not be included. Reconciliation must
-     * be performed when converting a clean page to a dirty page, which can increase latency.
+     * be performed when converting a clean page to a dirty page, which can increase latency. This
+     * check is bypassed if the session is configured with a debug option to evict the page when it
+     * is released and no longer needed.
      */
-    if (!F_ISSET(session, WT_SESSION_EVICTION))
+    if (!F_ISSET(session, WT_SESSION_EVICTION) && !F_ISSET(session, WT_SESSION_DEBUG_RELEASE_EVICT))
```
[the remark this answers is not in the document]
Comment (etienneptl) [reply src/evict/evict_page.c]: @kommiharibabu, should we backport WT-8208?

File: test/suite/test_eviction02.py [hunk 2]
```diff
@@ -26,89 +26,77 @@
...
-            if i > 0:
-                self.check_stat(prev_obsolete_tw_value, current_obsolete_tw_value)
-            prev_obsolete_tw_value = current_obsolete_tw_value
+            # Bump the oldest timestamp to make some of the previously inserted data globally
+            # visible. This makes any time window informaton associated with that data obsolete and
+            # eligible for cleanup.
+            self.conn.set_timestamp('oldest_timestamp=' + self.timestamp_str(num_keys // 2))
 
-            self.conn.set_timestamp('oldest_timestamp=' + self.timestamp_str(nrows * (i + 1)))
+            # Eviction should perform clean eviction here which triggers the assessement of obsolete
```
Comment (etienneptl) [inline test/suite/test_eviction02.py:76]: This is not true without the debug option. We will evict the page without reviewing the time window information.

**Comment by etienneptl** [comment]
I am worried that the test eviction03 is failing, this is the only proof we have that the disk is actually decreasing in size but I am not sure if more changes are needed to make this data accurate.

File: test/suite/test_eviction02.py (hunk 2 above)
Comment (etienneptl) [reply test/suite/test_eviction02.py]: Removed that function in 9a20093a4914788383fbc5011a4cbfd5e6448c5e

**Comment by etienneptl** [comment]
PR failure does not seem to be related:
```
file:test_wt7989_compact_checkpoint.wt, eviction-server: __wt_gen_active, 253: generation < oldest
```

**Review by kommiharibabu — APPROVED** [review]
LGTM

**Pull request merged**